vehicle/Jeep.h

```diff
@@ -0,0 +1,17 @@
+#ifndef JEEP_H
+#define JEEP_H
+
+#include "ICar.h"
+
+template <typename EngineType, typename GearType>
```

Comment:
are you sure it's the best solution? why not simply use interfaces?

Reply:
template removed